hubclient/apitoken-client.go

```diff
@@ -0,0 +1,81 @@
+package hubclient
+
+import (
+	"encoding/json"
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+type BearerTokenResponse struct {
+	BearerToken           string `json:"bearerToken"`
+	ExpiresInMilliseconds int64  `json:"expiresInMilliseconds"`
+}
+
+func NewWithApiToken(baseURL string, apiToken string, debugFlags HubClientDebug, timeout time.Duration) (*Client, error) {
```

Comment:
Do we have some sort of central configration here?
If yes, baseURL, debugFlags and timeout can be taken from there.

Reply:
I'm following the example of NewWithSession() and NewWithToken() in client.go here. This is a constructor for a new Client object, so there's nowhere to get the baseUrl.

We could instead have this be similar to Login() in login_client.go and be a method called on an existing object. We'd still need to provide a new constructor, though, since neither NewWithSession() (no need for cookies) nor NewWithToken() (no token yet) applies here.

Reply:
Given your answer, it's ok